plugins/modules/purefa_volume.py

```diff
@@ -1048,11 +1050,26 @@ def copy_from_volume(module, array):
     )
 
 
+def pg_check(module, array):
+    purity_version = array.get()["version"]
```

Comment:
Use the REST version instead of the Purity version for consistency. The REST version you need for 6.3.4 is 2.16, already defined as `DEFAULT_API_VERSION`.
The error message can reference the Purity version as that is what users will recognize.
You get the current rest version with `api_version = array.get_rest_version()`

Reply:
Modified according to suggestions.